Detectors/MUON/MCH/Raw/Decoder/include/MCHRawDecoder/DataDecoder.h

```diff
@@ -0,0 +1,82 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file    DataDecoder.h
+/// \author  Andrea Ferrero
+///
+/// \brief Definition of a data processor to run the raw decoding
+///
+
+#include <gsl/span>
+#include <unordered_set>
+
+#include "Headers/RDHAny.h"
+#include "MCHBase/Digit.h"
+#include "MCHRawDecoder/OrbitInfo.h"
+#include "MCHRawDecoder/PageDecoder.h"
+
+namespace o2
+{
+namespace mch
+{
+namespace raw
+{
+
+using RdhHandler = std::function<void(o2::header::RDHAny*)>;
+
+// custom hash for OrbitInfo objects
+struct OrbitInfoHash {
+  std::size_t operator()(OrbitInfo const& info) const noexcept
+  {
+    return std::hash<uint64_t>{}(info.get());
+  }
+};
+
+//_________________________________________________________________
+//
+// Data decoder
+//_________________________________________________________________
+class DataDecoder
+{
+ public:
+  DataDecoder(SampaChannelHandler channelHandler, RdhHandler rdhHandler, std::string mapCRUfile, std::string mapFECfile, bool ds2manu, bool verbose);
+
+  void reset();
+  void decodeBuffer(gsl::span<const std::byte> page);
+
+  std::vector<o2::mch::Digit>& getOutputDigits() { return mOutputDigits; }
```

Comment:
those two `getxxx` should return const references and be const functions, e.g. `const std::vector<o2::mch::Digit>& getOutputDigits() const { return ... }`

Reply:
Done